articles/spotify-python-part-1/spotipy_spotify_python.md

```diff
@@ -0,0 +1,136 @@
+
+
+## Step 2: Creating a python environment 
+
+
+A good developer practice is to have an organized file system. If this is your first python project consider making a directory called PythonScripts (you can name it whatever you want). Inside that directory create a folder called spotipy_tutorial.
+
+Now run the following command to recreate a virtual environment. **Do not skip this step**. If you already have virtualenv installed run the line:
+
+`~/PythonScripts/spotipy_tutorial >> virtualenv venv`
+
+If you have never installed a virtual environment before or the line above doesn't work, do the following: 
+
+`~/PythonScripts/spotipy_tutorial >> python3 -m pip3 install --user virtualenv`
+
+`~/PythonScripts/spotipy_tutorial >> virtualenv venv`
+
+
+**Why create a venv?** 
+
+Creating a virtual environment is usually overlooked by beginner programmers but this is a bad mistake to make. New versions of Python are released almost every year and this can cause problems when you want to return to a project. Modules start to become incompatible with the new versions of Python and your code begins to break. I speak from experience,  it is much harder to fix your program after it breaks (especially after 6+ months of not reading your code). Additionally, some modules can have bugs that can do some nasty stuff to your system. By creating a self-contained directory you can avoid  stress and worry. 
+
+Note: Spotipy is a well-built module and has an active team supporting development, so you don't have to worry about catastrophic bugs, but it is still better to get in the habit of creating virtual environments for your project. 
+
+After the venv was created, run this command to start your virtual environment. 
+
+`~/PythonScripts/spotipy_tutorial >> source venv/bin/activate`
+
+And to deactivate run:
+
+`~/PythonScripts/spotipy_tutorial >> deactivate`  
+
+Now that you have a Python environment set up, you should install spotipy.
+
+Run the following command in your terminal window with the active virtual environment:
+
+`~/PythonScripts/spotipy_tutorial >> pip install spotipy`
+
+## Step 3: Authenticate and Run 
+
+You are going to need to client ID, client Secret, and redirect url for this part. You can either set these values as environment variables:
+
+`export SPOTIPY_CLIENT_ID='your-spotify-client-id'`
+
+`export SPOTIPY_CLIENT_SECRET='your-spotify-client-secret'`   
+
+`export SPOTIPY_REDIRECT_URI='your-spotify-redirect-url'`
+
+I recommend you create a file called cred.py and store your values inside it. It goes without saying that this file should not be committed to a repository. 
+
+Create a cred.py file and put the following in your file: 
+
+    client_ID='your-spotify-client-id'
+    client_SECRET='your-spotify-client-secret'   
+    redirect_url='your-spotify-redirect-url'
+
+Now create a file main.py and add the following boilerplate code: 
+
+    import spotipy
```

Comment:
```py
    import spotipy
    from spotipy.oauth2 import SpotifyOAuth
    import cred 
```

Reply:
Refer to this [article ](https://www.freecodecamp.org/news/how-to-format-code-in-markdown/#)for more details.